hass_nabucasa/utils.py

```diff
@@ -24,8 +25,12 @@ def utc_from_timestamp(timestamp: float) -> dt.datetime:
 
 def parse_date(dt_str: str) -> dt.date | None:
     """Convert a date string to a date object."""
+    if (match := DATE_STR_FORMAT.match(dt_str)) is None:
+        return None
+
+    [year, month, day] = [int(group) for group in match.groups()]
     try:
-        return dt.datetime.strptime(dt_str, DATE_STR_FORMAT).date()
+        return dt.datetime(year, month, day).date()
     except ValueError:  # If dt_str did not match our format
```

Comment:
An alternative to this is that we can run `locale.setlocale(locale.LC_ALL, "")` before using `strptime`